src/router.zig

```diff
@@ -0,0 +1,77 @@
+const std = @import("std");
+const zap = @import("zap.zig");
+
+const Allocator = std.mem.Allocator;
+const RouterError = error{
+    AlreadyExists,
+    EmptyPath,
+};
+
+// inline closure for RequestFn with self argument
+pub inline fn RequestHandler(self: anytype, func: *const fn (@TypeOf(self), zap.Request) void) *const fn (zap.Request) void {
```

Comment:
maybe, if it should be available under zap namespace, move this into its own file and import in zap usingnamespace.